ui/src/components/stepforms/schemas/dategranularity.ts

```diff
@@ -18,9 +18,10 @@ export default {
     },
     granularity: {
       type: 'string',
-      enum: ['year', 'quarter', 'month', 'isoWeek', 'week', 'day'],
+      minLength: 1,
       title: 'Date granularity to apply',
-      description: 'At what granularity will the date be truncated.',
+      description:
+        "At what granularity will the date be truncated. Should be 'year', 'quarter', 'month', 'isoWeek', 'week', 'day' or a variable.",
```

Comment:
```suggestion
        "The granularity to which the date will be truncated. Should be one of 'year', 'quarter', 'month', 'isoWeek', 'week' or 'day' .",
```

The variable picker will be available through the UI. "or a variable" sounds like it is possible to invent new granularities, doesn't it ?

Reply:
[8fa50c0](https://github.com/ToucanToco/weaverbird/pull/2413/commits/8fa50c0fb667d3734a88fa80bd4d85d3d967113c)